Libraries/PeriphDrivers/Source/SPIMSS/spimss_reva.c

```diff
@@ -231,6 +242,197 @@ int MXC_SPIMSS_RevA_MasterTrans(mxc_spimss_reva_regs_t *spi, spimss_reva_req_t *
+    memset(&dma_adv_config_tx, 0, sizeof(dma_adv_config_tx));
+    memset(&dma_adv_config_rx, 0, sizeof(dma_adv_config_rx));
+
+    // We use SPIMSS master trans function to send data from
+    if (E_NO_ERROR != MXC_SPIMSS_RevA_TransSetup(spi, req, 1)) {
+        return -1;
+    }
+
+    //Setting dmas fifo levels for tx and rx in spimss.
+    spi->dma &= ~(7 << MXC_F_SPIMSS_DMA_RX_FIFO_LVL_POS);
+    spi->dma &= ~(7 << MXC_F_SPIMSS_DMA_TX_FIFO_LVL_POS);
+
+    //Initialization of DMA
+    MXC_DMA_Init();
+
+    if ((true == states[spi_num].auto_dma_handlers) && (0 > states[spi_num].channelTx)) {
+        // Getting the available tx channel id.
+        tx_channel_id = MXC_DMA_AcquireChannel();
+        if (tx_channel_id < 0) {
+            return -1;
+        }
+        MXC_SPIMSS_RevA_SetTXDMAChannel(spi, tx_channel_id);
+
+        //Enable NVIC interrupt of the dma channel which will be used as TX channel.
+        //NVIC_EnableIRQ(MXC_DMA_CH_GET_IRQ(tx_channel_id));
+        MXC_NVIC_SetVector(MXC_DMA_CH_GET_IRQ(tx_channel_id), MXC_DMA_Handler);
+    } else {
+        tx_channel_id = MXC_SPIMSS_RevA_GetTXDMAChannel(spi);
+        if (tx_channel_id < 0) {
+            return E_BAD_STATE;
+        }
+    }
+
+    if ((true == states[spi_num].auto_dma_handlers) && (0 > states[spi_num].channelRx)) {
+        rx_channel_id = MXC_DMA_AcquireChannel();
+        if (rx_channel_id < 0) {
+            return -1;
+        }
+        MXC_SPIMSS_RevA_SetRXDMAChannel(spi, rx_channel_id);
+
+        //Enable NVIC interrupt of the dma channel which will be used as RX channel.
+        //NVIC_EnableIRQ(MXC_DMA_CH_GET_IRQ(rx_channel_id));
+        MXC_NVIC_SetVector(MXC_DMA_CH_GET_IRQ(rx_channel_id), MXC_DMA_Handler);
+    } else {
+        rx_channel_id = MXC_SPIMSS_RevA_GetRXDMAChannel(spi);
+    }
+
+    states[spi_num].req = req;
+
+    // TX dma configuration settings.
+    dma_config_tx.ch = tx_channel_id;
+    dma_config_tx.reqsel = MXC_DMA_REQUEST_SPI1TX;
+    dma_config_tx.dstwd = MXC_DMA_WIDTH_HALFWORD;
+    dma_config_tx.srcwd = MXC_DMA_WIDTH_HALFWORD;
+    dma_config_tx.dstinc_en = 0; // Ignored since the destination is set to SPIMSS TX fifo.
+    dma_config_tx.srcinc_en = 1;
+
+    // advanced config for tx channel
+    dma_adv_config_tx.ch = tx_channel_id;
+    dma_adv_config_tx.burst_size = 2;
+
+    // advanced config for rx channel
+    dma_adv_config_rx.ch = rx_channel_id;
+    dma_adv_config_rx.burst_size = 2;
+
+    dma_config_rx.ch = rx_channel_id;
+    dma_config_rx.reqsel = MXC_DMA_REQUEST_SPI1RX;
+    dma_config_rx.dstwd = MXC_DMA_WIDTH_HALFWORD;
+    dma_config_rx.srcwd = MXC_DMA_WIDTH_HALFWORD;
+    dma_config_rx.dstinc_en = 1;
+    dma_config_rx.srcinc_en = 0; // Ignored since the source is set to SPIMSS RX fifo.
+
+    //Setting TX configuration values for this spi transaction.
+    srcdst_config_tx.ch = tx_channel_id;
+    srcdst_config_tx.len = sent_byte_len;
+    srcdst_config_tx.source = (void *)req->tx_data;
```

Comment:
In general, casting to a void pointer shouldn't be necessary

Reply:
(void*) casting is removed.